### DISCUSSION TITLE
Using multiple persistence in slices pattern is causing inconsistency.

### DISCUSSION THREAD

**Discussion by terr0raid** [pmndrs/zustand#2437] [Bug report] [answered]
Hi, i am sorry for not gonna share a detailed format of issue. If you want to more information, i can share.

My problem is when i am using multiple persistence in one bounded context, zustand is only rehydrating first persistence where is created in store list.

```js
import { create } from 'zustand'

import { createGpsSlice, GpsSlice } from './gps-slice'
import { createUserSlice, UserSlice } from './user-slice'
import { createPreferencesSlice, PreferencesSlice } from './preferences-slice'
import { persist } from 'zustand/middleware'

export interface Store extends GpsSlice, UserSlice, PreferencesSlice {
	hydrateStore(initialStore: any): void
}

export const useBoundedStore = create<Store>((...args) => ({
	...createGpsSlice(...args),
	...persist(createUserSlice, {
		name: 'user',
		partialize: s => ({
			selectedMembership: s.selectedMembership,
		}),
	})(...args),
	...persist(createPreferencesSlice, {
		name: 'preferences',
		partialize: s => ({
			sideBarStatus: s.sideBarStatus,
		}),
	})(...args),
	hydrateStore: (data: any) => {
		const [set] = args
		set(data)
	},
}))
```
The bounded context state is returning this. But it should has two more field named setSideBarStatus and sideBarStatus. The point i want to mention is when if i change the order that two slice, the context state involves only first slice.
  
![image](https://github.com/pmndrs/zustand/assets/72807239/e937099c-456b-446a-8288-07f0784a8053)

##Versions
zustand: "^4.5.1"
next: "14.1.0"

**Comment by dbritto-dev** [comment, marked as answer]
@terr0raid persist shouldn't be use on slices but only at store level like any middleware. Ref: https://docs.pmnd.rs/zustand/guides/slices-pattern#adding-middlewares

**Reply by terr0raid** [reply]
Oh, I didn't know that, I'll try. Thank you

**Reply by dbritto-dev** [reply]
@terr0raid btw, you can read about that here -> https://docs.pmnd.rs/zustand/guides/slices-pattern#adding-middlewares but let me know if there's something that's not clear or there's something we need to improve, we are open to suggestions

**Reply by terr0raid** [reply]
Thank you for help, I've solved my mistake.

```js
export const useBoundedStore = create<Store>()(
	persist(
		(...a) => ({
			...createGpsSlice(...a),
			...createUserSlice(...a),
			...createPreferencesSlice(...a),
			hydrateStore: (data: any) => {
				const [set] = a
				set(data)
			},
		}),
		{
			name: 'bounded-store',
			partialize: s => ({
				sideBarStatus: s.sideBarStatus,
				selectedMembership: s.selectedMembership,
			}),
		}
	)
)
```
